src/main/java/org/broadinstitute/hellbender/tools/htsgetreader/HtsgetResponse.java

```diff
@@ -110,4 +94,25 @@ public HtsgetFormat getFormat() {
     public String getMd5() {
         return this.md5;
     }
+
+    /**
+     * Lazily generates an InputStream over this response's data from the concatenation of the InputStreams from
+     * each of the response's data blocks
+     * @return InputStream over this response's data
+     */
+    public InputStream getDataStream() {
+        return new SequenceInputStream(new Enumeration<InputStream>() {
```

Comment:
Huh, I don't remember ever seeing Enumeration before.  Seems like the right tool for the job though.

Reply:
It's exactly the same interface as Iterator but SequenceInputStream only accepts Enumeration so this was the simplest way to get an InputStream that lazily requests blocks from the htsget server